### PR TITLE
Add service management helpers

### PR DESCRIPTION
**Pull request by Jyrno42**
Provide management helpers for the following unix service daemon utilities:
- upstart
- systemd
- supervisor

### CODE REVIEW

File: hammer/service_helpers.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+
+def manage_service(names, action):
+    """Perform `action` on services
+
+    :param names: Can be a list of services or a name of a single service to control
+    :param action: Action that should be executed for the given services
+    """
+    if not isinstance(names, (list, tuple)):
+        names = [names, ]
```
Comment (rivol) [inline hammer/service_helpers.py:124]: nitpick: superfluous comma

File: hammer/service_helpers.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    if daemon_type == 'supervisor':
+        # Ensure configuration files are reloaded
+        manage_service('supervisorctl', 'reread')
+
+    elif daemon_type == 'systemd':
+        # Ensure configuration files are reloaded
+        manage_service('systemctl', 'daemon-reload')
+
+        # Ensure services are started on startup
+        manage_service([x[1] for x in services], 'enable')
```
Comment (rivol) [inline hammer/service_helpers.py]: Should there be `x[0]` here?

Or rather `[target_name for target_name, file_data in services]`

File: hammer/service_helpers.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+DAEMON_TYPES = {
+    'systemd': {
+        'daemon_cmd': 'systemctl %(action)s %(name)s',
+        'target_dir': '/etc/systemd/system',
+        'file_extension': 'service',
+    },
+    'supervisor': {
+        'daemon_cmd': 'supervisorctl %(action)s %(name)s',
+        'target_dir': ('supervisor_target_dir', '/etc/supervisord/conf.d/'),
```
Comment (rivol) [inline hammer/service_helpers.py]: Lets make the target dir part of env, e.g. service_daemon_target_dir

**Comment by Jyrno42** [comment]
@rivol Updated PR, please let me know if it is OK, so i can squash the commits before the merge

**Pull request merged**